### PR TITLE
Post-Continental QA work

### PR DESCRIPTION
**Pull request by jensck**
This review is probably best taken commit by commit, if you're up for that.  As intended, it's mostly lots of bug fixes and error reporting improvements.

### CODE REVIEW

File: commonui/src/main/java/io/particle/commonui/Devices.kt [hunk 1]
```diff
@@ -33,35 +33,38 @@ private data class ProductImageAndName(@StringRes val name: Int, @DrawableRes va
...
             BLUZ,
             DIGISTUMP_OAK,
-            OTHER -> pin(R.string.product_name_unknown, R.drawable.product_image_unknown)
+            OTHER -> pian(R.string.product_name_unknown, R.drawable.product_image_unknown)
         }
     }
 
 
-private fun pin(@StringRes name: Int, @DrawableRes image: Int): ProductImageAndName {
+private fun pian(@StringRes name: Int, @DrawableRes image: Int): ProductImageAndName {
```
Comment (RaimundasSakalauskas) [inline commonui/src/main/java/io/particle/commonui/Devices.kt]: Was this rename intentional?

Comment (jensck) [reply commonui/src/main/java/io/particle/commonui/Devices.kt]: Yes. It's just a tiny, private "macro"-type function to keep everything above in single lines, to make it more readable.  I renamed it to fit the initialism, "PIAN" (`ProductImageAndName`).  Calling it "pin" felt misleading.

Not the sort of naming I'd usually use, but I thought it was the best trade-off here.

**Comment by jensck** [comment]
Reviewed this with Raimis before Partivision; merging
